std/container/array.d

```diff
@@ -142,26 +142,26 @@ private struct RangeT(A)
 
         E moveFront()
         {
-            version (assert) if (empty || _a >= _outer.length) throw new RangeError();
+            assert(!empty || _a < _outer.length);
```

Comment:
Unless I'm having a dumb moment, this should use `&&`. `!(foo || bar)` = `!foo && !bar`. Same mistake throughout.